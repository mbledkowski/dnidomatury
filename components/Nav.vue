<template>
  <nav>
    <nuxt-link to="/calendar">
      <img
        src="/icons/calendar.svg"
        alt="calendar"
        class="sub nav-icon invert"
      />
      <p class="main">{{ $t('calendar') }}</p>
    </nuxt-link>
    <nuxt-link to="/">
      <img src="/icons/timer.svg" alt="timer" class="sub nav-icon invert" />
      <p class="main">{{ $t('timer') }}</p>
    </nuxt-link>
    <nuxt-link to="/news">
      <img src="/icons/news.svg" alt="news" class="sub nav-icon invert" />
      <p class="main">{{ $t('news') }}</p>
    </nuxt-link>
    <nuxt-link to="/settings">
      <img
        src="/icons/settings.svg"
        alt="settings"
        class="main nav-icon invert"
      />
      <p class="sub">{{ $t('settings.title') }}</p>
    </nuxt-link>
  </nav>
</template>
<script>
export default {
  name: 'MainNav',
}
</script>
<style lang="scss">
nav {
  @apply flex flex-row justify-evenly items-center mx-auto text-white bg-black rounded-none md:rounded-lg h-16 md:h-8 text-lg font-bold w-full md:w-96 lowercase fixed bottom-0 z-10 md:static md:my-8;
  > a {
    @apply w-1/4 md:w-auto text-center;
    > img {
      @apply mx-auto;
    }
    > p {
      @apply truncate;
    }
  }
}
.sub {
  @apply visible;
  @apply md:hidden;
}
</style>
