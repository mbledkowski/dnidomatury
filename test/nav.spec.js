import { shallowMount } from '@vue/test-utils'
import Nav from '@/components/Nav.vue'

describe('Nav', () => {
  const wrapper = shallowMount(Nav, {
    stubs: {
      'nuxt-link': true,
      'any-other-child': true,
    },
    mocks: {
      // Always returns the input
      $t: (i) => i,
      localePath: (i) => i,
    },
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
  it('renders correctly with links', () => {
    wrapper.setProps({
      links: [
        {
          name: 'timer',
          path: '/',
        },
        {
          name: 'settings',
          path: '/settings',
        },
      ],
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
