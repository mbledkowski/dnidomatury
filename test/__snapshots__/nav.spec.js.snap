// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<nav>
  <nuxt-link-stub
    to="/calendar"
  >
    <img
      alt="calendar"
      class="sub nav-icon invert"
      src="/icons/calendar.svg"
    />
     
    <p
      class="main"
    >
      calendar
    </p>
  </nuxt-link-stub>
   
  <nuxt-link-stub
    to="/"
  >
    <img
      alt="timer"
      class="sub nav-icon invert"
      src="/icons/timer.svg"
    />
     
    <p
      class="main"
    >
      timer
    </p>
  </nuxt-link-stub>
   
  <nuxt-link-stub
    to="/news"
  >
    <img
      alt="news"
      class="sub nav-icon invert"
      src="/icons/news.svg"
    />
     
    <p
      class="main"
    >
      news
    </p>
  </nuxt-link-stub>
   
  <nuxt-link-stub
    to="/settings"
  >
    <img
      alt="settings"
      class="main nav-icon invert"
      src="/icons/settings.svg"
    />
     
    <p
      class="sub"
    >
      settings.title
    </p>
  </nuxt-link-stub>
</nav>
`;

exports[`Nav renders correctly with links 1`] = `
<nav>
  <nuxt-link-stub
    to="/calendar"
  >
    <img
      alt="calendar"
      class="sub nav-icon invert"
      src="/icons/calendar.svg"
    />
     
    <p
      class="main"
    >
      calendar
    </p>
  </nuxt-link-stub>
   
  <nuxt-link-stub
    to="/"
  >
    <img
      alt="timer"
      class="sub nav-icon invert"
      src="/icons/timer.svg"
    />
     
    <p
      class="main"
    >
      timer
    </p>
  </nuxt-link-stub>
   
  <nuxt-link-stub
    to="/news"
  >
    <img
      alt="news"
      class="sub nav-icon invert"
      src="/icons/news.svg"
    />
     
    <p
      class="main"
    >
      news
    </p>
  </nuxt-link-stub>
   
  <nuxt-link-stub
    to="/settings"
  >
    <img
      alt="settings"
      class="main nav-icon invert"
      src="/icons/settings.svg"
    />
     
    <p
      class="sub"
    >
      settings.title
    </p>
  </nuxt-link-stub>
</nav>
`;
